script.js

```diff
@@ -38,7 +44,8 @@ headersReceivedListener = (requestDetails) => {
   let filter = browser.webRequest.filterResponseData(requestDetails.requestId);
 
   filter.ondata = event => {
-    const origin = extractHostname(!requestDetails.originUrl ? requestDetails.url : requestDetails.originUrl);
+    // If Incognito request we do not track the domain origin
+    const origin = (requestDetails.incognito ? CONST_INCOGNITO : extractHostname(!requestDetails.originUrl ? requestDetails.url : requestDetails.originUrl));
```

Comment:
It's easier to read like this I think (without negation)

```suggestion
    const origin = (requestDetails.incognito ? CONST_INCOGNITO : extractHostname(requestDetails.originUrl ? requestDetails.originUrl : requestDetails.url));
```